### PR TITLE
Set TPU VM the default option

### PR DESCRIPTION
**Pull request by infwinston**
This PR changes the default for TPU from TPU node to TPU VM as the former is now being deprecated by GCP.
for example, the tpu-v4 arch doesn't support TPU node ([ref](https://cloud.google.com/tpu/docs/v4-users-guide))
```
For Cloud TPU v4, only the TPU VM architecture is supported, so all TPUs referenced in this document use the TPU VM architecture.
```

Tested (run the relevant ones):

- [ ] Disable tpu node for v4
- [ ] Smoke tests
- [ ] Docs
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

**Comment by Michaelvll** [comment]
@infwinston Any updates?

**Comment by Michaelvll** [comment]
We need someone to take over this PR to get it in. This is requested by our user.

**Comment by Michaelvll** [comment]
Tested:
- [ ] on master: `sky launch examples/tpu/tpu_app.yaml -c test-tpu-node`; on this PR: `sky exec test-tpu-node examples/tpu/tpu_app.yaml`; `sky launch examples/tpu/tpu_app.yaml -c test-tpu-node`
- [ ] `sky launch examples/tpu/tpuvm_mnist.yaml`

File: sky/backends/cloud_vm_ray_backend.py [hunk 1]
```diff
@@ -2666,6 +2666,19 @@ def __setstate__(self, state):
         if version < 6:
             state['cluster_name_on_cloud'] = state['cluster_name']
 
+        if version < 7:
+            launched_resources = state['launched_resources']
+
+            # Backward compatibility: we change the default value for TPU VM to
```
Comment (concretevitamin) [inline sky/backends/cloud_vm_ray_backend.py:2672]: QQ: why handle here vs. in https://github.com/skypilot-org/skypilot/blob/88562fe67aa11db771b546e24a7fd04015309e6d/sky/resources.py#L47?

File: sky/clouds/gcp.py [hunk 2]
```diff
@@ -810,7 +809,6 @@ def need_cleanup_after_preemption(self,
         # See: https://cloud.google.com/tpu/docs/preemptible#tpu-vm
 
         # pylint: disable=import-outside-toplevel
```
Comment (concretevitamin) [inline sky/clouds/gcp.py:811]: this line can be removed

**Review by concretevitamin — COMMENTED** [review]
Thanks @Michaelvll @infwinston. Quick question.

**Comment by concretevitamin** [comment]
May want to update https://skypilot.readthedocs.io/en/latest/reference/yaml-spec.html and https://skypilot.readthedocs.io/en/latest/reference/tpu.html too.

File: sky/clouds/gcp.py (hunk 2 above)
Comment (Michaelvll) [reply sky/clouds/gcp.py]: Good catch! Removed. Thanks!

File: sky/backends/cloud_vm_ray_backend.py (hunk 1 above)
Comment (Michaelvll) [reply sky/backends/cloud_vm_ray_backend.py]: Good catch! Moved it to resources instead. : )

**Comment by Michaelvll** [comment]
Thanks @concretevitamin for the review! Updated the docs. PTAL

File: docs/source/reference/yaml-spec.rst [hunk 3]
```diff
@@ -139,19 +139,20 @@ Available fields:
       #   To request a TPU node:
       #     accelerator_args:
       #       tpu_name: ...
+      #       tpu_vm: False
       #
       #   To request a TPU VM:
       #     accelerator_args:
-      #       tpu_vm: True
+      #       tpu_vm: True (optional, default: True)
```
Comment (concretevitamin) [inline docs/source/reference/yaml-spec.rst:146]: nit: Reorder these above L139?

File: sky/resources.py [hunk 4]
```diff
@@ -1340,4 +1340,14 @@ def __setstate__(self, state):
                 state['_ports'] = resources_utils.simplify_ports(
                     [str(port) for port in original_ports])
 
+        if version < 14:
+            # Backward compatibility: we change the default value for TPU VM to
+            # True in version 14 (#1758), so we need to explicitly set it to
+            # False when loading the old handle.
+            if state.get('accelerators', '').startswith('tpu'):
```
Comment (concretevitamin) [inline sky/resources.py:1347]: Can accelerators be a dict like
```
accelerators:
  tpu-v2-8: 1
```
?

**Review by concretevitamin — APPROVED** [review]

**Comment by Michaelvll** [comment]
Tested:
- [ ] master branch: `sky launch -c test-tpu-node test.yaml`; this PR: `sky exec test-tpu-node test-with-acc-args.yaml`
```yaml
resources: 
  accelerators: tpu-v2-8

run: |
  echo "Hello World"
```

File: sky/resources.py (hunk 4 above)
Comment (Michaelvll) [reply sky/resources.py]: Ahh, good catch! Fixed and now tested it. : )

**Pull request merged**